synapse/lib/hiveauth.py

```diff
@@ -574,6 +590,7 @@ def getRules(self, gateiden=None):
         return list(gateinfo.get('rules', ()))
 
     async def setRules(self, rules, gateiden=None, nexs=True):
+        reqValidRules(rules)
         return await self._setRulrInfo('rules', rules, gateiden=gateiden, nexs=nexs)
 
     async def addRule(self, rule, indx=None, gateiden=None, nexs=True):
```

Comment:
You need to protect do validation around addrule.

`reqValidRules((rule,))` would do it.